### PR TITLE
PDI-1813: Export block HCL generation for PingFederate

### PR DESCRIPTION
**Pull request by erikostien-pingidentity**
- Update Option Objects variable names to contain the service name to
distinguish between PingOne options and PingFederate Options.
- Update Cobra Subcommand `platform export` to support the new export
service PingFederate. Add additional examples for PingFederate export
parameters. Break out cobra flag configuration by service, and add
PingFederate flags. Mark each PingFederate Auth method as mutually
exclusive for the user to avoid authentication confusion.
- Update cobra root command root.go to fail if user provided config file
does not exist.
- Add profile config enum type ENUM_STRING_SLICE for PF scopes and
PF pem files. Support these slices in set_internal.go and validate.go.
- Update export_internal.go to initialize both PingOne go client and
PingFederate go client when needed. Update getExportableConnectors
to use instance variables depending on the service connector specified.
- Break out SDKClientInfo into PingOneClientInfo and PingFederateClientInfo
for different services.
- Create pingfederate_connector.go to export PF API resources.
- Add first PF resource for export, pingfederate_authentication_api_application.go
- Add server-profiles directory to hold PingFederate server-profile for
container starting and testing.

### CODE REVIEW

File: .github/workflows/code-analysis-lint-test.yaml [hunk 1]
```diff
@@ -143,4 +153,4 @@ jobs:
       - uses: hashicorp/setup-terraform@v3
         with:
           terraform_wrapper: false
-      - run: make test
+      - run: mkdir -p $HOME/.pingidentity/ && echo "$PING_IDENTITY_CONFIG" >> $HOME/.pingidentity/config && grep -E "[a-zA-Z]" $HOME/.pingidentity/config || exit 1 && make starttestcontainer && make test && make removetestcontainer
```
Comment (henryrecker-pingidentity) [inline .github/workflows/code-analysis-lint-test.yaml:156]: What is the grep doing here?

File: cmd/platform/export.go [hunk 2]
```diff
@@ -27,66 +27,225 @@ func NewExportCommand() *cobra.Command {
...
+		Option: profiles.PingFederateAdminApiPathOption,
+		Flag:   cmd.Flags().Lookup(profiles.PingFederateAdminApiPathOption.CobraParamName),
+	})
+	profiles.AddEnvVarBinding(profiles.PingFederateAdminApiPathOption)
+
+	// Require both HTTPS host and admin API path flags to be used together
+	cmd.MarkFlagsRequiredTogether(profiles.PingFederateHttpsHostOption.CobraParamName, profiles.PingFederateAdminApiPathOption.CobraParamName)
+
+	// X-Bypass-External-Validation header flag
+	cmd.Flags().Bool(profiles.PingFederateXBypassExternalValidationHeaderOption.CobraParamName, false, fmt.Sprintf("Header value in request for PingFederate. The connection test will be bypassed when set to true.  Also configurable via environment variable %s", profiles.PingFederateXBypassExternalValidationHeaderOption.EnvVar))
```
Comment (henryrecker-pingidentity) [inline cmd/platform/export.go:144]: "The connection test" sounds a little unclear, might be good to clarify that it's PF's own connection tests that get skipped, not something that pingctl is testing

File: internal/commands/platform/export_internal_test.go [hunk 3]
```diff
@@ -2,143 +2,569 @@ package platform_internal
...
+        clientcredentialsauth:
+            clientid: ""
+            clientsecret: ""
+            scopes: ""
+            tokenurl: ""
+        httpshost: "%s"`,
+		os.Getenv(profiles.PingFederateHttpsHostOption.EnvVar)))
+
+	expectedErrorPattern := `^failed to initialize PingFederate API client\. none of the following sets of authentication configuration values are set: OAuth2 client credentials \(client ID, client secret, token URL\), Access token, or Basic Authentication credentials \(username, password\)\. configure these properties via parameter flags, environment variables, or the tool\'s configuration file \(default: \$HOME/\.pingctl/config\.yaml\)$`
+	//expectedErrorPattern := `^failed to initialize pingfederate API client\. the pingfederate https host configuration value is not set: configure this property via parameter flags, environment variables, or the tool\'s configuration file \(default: \$HOME/\.pingctl/config\.yaml\)$`
```
Comment (henryrecker-pingidentity) [inline internal/commands/platform/export_internal_test.go:267]: Is this supposed to be here?

**Review by henryrecker-pingidentity — APPROVED** [review]

File: .github/workflows/code-analysis-lint-test.yaml (hunk 1 above)
Comment (erikostien-pingidentity) [reply .github/workflows/code-analysis-lint-test.yaml:156]: I believe it is checking the config file was populated. This logic is from the PF provider makefile.

File: cmd/platform/export.go (hunk 2 above)
Comment (erikostien-pingidentity) [reply cmd/platform/export.go:144]: Updated

File: internal/commands/platform/export_internal_test.go (hunk 3 above)
Comment (erikostien-pingidentity) [reply internal/commands/platform/export_internal_test.go]: removed

**Pull request merged**